### PR TITLE
ci: Use cri-containerd release tarball.

### PR DESCRIPTION
**Pull request by jcvenegas**
Lets use cri-container release tarball. If
the tarball installation fails build a release
tarball from source and install it.

Fixes: #1399

### CODE REVIEW

**Comment by jcvenegas** [comment]
Depends-on: github.com/kata-containers/runtime#1465

**Comment by chavafg** [comment]
containerd tests failing:
```
16:46:34 INFO: Containerd checks
16:46:34 bash integration/containerd/cri/integration-tests.sh
16:46:34 INFO: Stop crio service
16:46:34 integration/containerd/cri/integration-tests.sh: line 154: pushd: /tmp/jenkins/workspace/kata-containers-tests-cri-containerd-PR/go/src/github.com/containerd/cri: No such file or directory
16:46:34 Makefile:105: recipe for target 'cri-containerd' failed
16:46:34 make: *** [cri-containerd] Error 1
16:46:34 Build step 'Execute shell' marked build as failure
```

**Comment by jcvenegas** [comment]
```
10:14:55 === RUN   TestImageLoad
10:15:00 E0403 16:15:00.704522   18401 remote_runtime.go:109] RunPodSandbox from runtime service failed: rpc error: code = Unknown desc = failed to get sandbox runtime: no runtime for "kata-runtime" is configured
10:15:00 --- FAIL: TestImageLoad (5.28s)
10:15:00     image_load_test.go:41: docker save image into tarball
10:15:00     image_load_test.go:53: make sure no such image in cri
10:15:00     image_load_test.go:60: load image in cri
10:15:00     image_load_test.go:65: make sure image is loaded
10:15:00     image_load_test.go:71: create a container with the loaded image
10:15:00     assertions.go:237: 
                        
	Error Trace:	image_load_test.go:74
10:15:00         
	Error:		Received unexpected error rpc error: code = Unknown desc = failed to get sandbox runtime: no runtime for "kata-runtime" is configured
```
Now failed  at

File: .ci/install_cri_containerd.sh [hunk 1]
```diff
@@ -8,67 +8,70 @@
 set -o errexit
 set -o nounset
 set -o pipefail
+set -o errtrace
+
+handle_error() {
```
Comment (jodh-intel) [inline .ci/install_cri_containerd.sh]: Nit: Could this go in `.ci/lib.sh` or `lib/common.bash`?

File: .ci/install_cri_containerd.sh [hunk 2]
```diff
@@ -8,67 +8,70 @@
...
 cri_repository="github.com/containerd/cri"
 
 # Flag to do tasks for CI
 CI=${CI:-""}
 
+# shellcheck source=./lib.sh
 source "${script_dir}/lib.sh"
+
+GOOS=$(go env GOOS)
+GOARCH=$(go env GOARCH)
```
Comment (jodh-intel) [inline .ci/install_cri_containerd.sh]: Again, this looks pretty generic so could go into one of the library scripts we source.

**Comment by jodh-intel** [comment]
Any update on this @jcvenegas?

**Comment by chavafg** [comment]
ping @jcvenegas

**Comment by jcvenegas** [comment]
```
18:22:45 [plugins]
18:22:45   [plugins.cri]
18:22:45     [plugins.cri.containerd]
18:22:45 	default_runtime_name = "io.containerd.kata.v2"
18:22:45       [plugins.cri.containerd.runtimes.io.containerd.kata.v2]
18:22:45         runtime_type = "io.containerd.kata.v2"
18:22:45         [plugins.cri.containerd.runtimes.io.containerd.kata.v2.options]
18:22:45           Runtime = "io.containerd.kata.v2"
18:22:45 [plugins.linux]
```

will return:
```
failed to create PodSandbox: rpc error: code = Unimplemented desc = unknown service runtime.v1alpha2.RuntimeService
18:23:20     Expected error:
18:23:20         <*status.statusError | 0xc00043e960>: {
18:23:20             Code: 12,
18:23:20             Message: "unknown service runtime.v1alpha2.RuntimeService",
18:23:20             Details: nil,
18:23:20         }
```

**Comment by jcvenegas** [comment]
@jodh-intel now I see some jobs failing with things related with tracing, are you familiar with that error or could give me some guidance what should I check?
```
00:07:53 Digest: sha256:51b901b653f4a4ca5dd50be9133c08dacf2d3eb092e507c213e7955a0e132297
00:07:53 Status: Downloaded newer image for jaegertracing/all-in-one:latest
00:07:54 ff81c9bf36097c0306b1b6af41eeaa3bd2b81c40bb8d3487cc4ca7fdb5377336
00:07:55 INFO: Enabling kata tracing on /usr/share/defaults/kata-containers/configuration.toml
00:07:55 INFO: Checking runtime spans
00:08:00 INFO: found 622 spans (across 9 traces)
00:08:00 WARNING: Found invalid parent span errors (attempt 0): Failed at 182: grep "invalid parent span"
00:08:00 INFO: found 622 spans (across 9 traces)
00:08:00 WARNING: Found invalid parent span errors (attempt 1): Failed at 182: grep "invalid parent span"
00:08:00 INFO: found 622 spans (across 9 traces)
00:08:00 WARNING: Found invalid parent span errors (attempt 2): Failed at 182: grep "invalid parent span"
00:08:00 ERROR: errors still detected after 3 attempts
00:08:01 jaeger
00:08:01 jaeger
00:08:01 INFO: Disabling kata tracing on /usr/share/defaults/kata-containers/configuration.toml
00:08:01 make: *** [tracing] Error 1
00:08:01 Failed at 40: sudo -E PATH="$PATH" bash -c "make test"
```

**Comment by jodh-intel** [comment]
Hi @jcvenegas - it's currently basically impossible to debug that sort of failure by looking at the CI logs only. I've raised #1593 to resolve that problem. However, generally if the tracing test fails, there are two main possibilities:

- Bad code change that created a new trace span and forgot to complete it.

  That's not likely here given that half of the CI's passed.

- Something crashed / got killed.

  That's likely what happened here so I'll try re-running the failing CI's to see if it happens again for the 18.04 one...

**Comment by jodh-intel** [comment]
Actually, for speed (for me :), since more CI's failed than passed...

/retest

File: .ci/install_cri_containerd.sh [hunk 3]
```diff
@@ -8,67 +8,75 @@
...
 
 # Flag to do tasks for CI
 CI=${CI:-""}
 
+# shellcheck source=./lib.sh
 source "${script_dir}/lib.sh"
+
+#Use cri contaienrd tarball format.
+#https://github.com/containerd/cri/blob/master/docs/installation.md#release-tarball
+CONTAIENRD_OS=$(go env GOOS)
```
Comment (jodh-intel) [inline .ci/install_cri_containerd.sh]: Typo: `CONTAINERD_OS`?

File: integration/containerd/cri/integration-tests.sh [hunk 4]
```diff
@@ -152,22 +164,19 @@ main() {
...
+	info "containerd(cri): testing using runtime: ${containerd_runtime_test}"
 
-	create_continerd_config "${runtime_cri}" "${runtime_bin}"
+	create_continerd_config "${containerd_runtime_test}"
 
 	info "containerd(cri): Running cri-tools"
 	sudo -E PATH="${PATH}:/usr/local/bin" \
 		FOCUS="runtime should support basic operations on container" \
+		RUNTIME="" \
+		SKIP="runtime should support execSync with timeout" \
```
Comment (jodh-intel) [inline integration/containerd/cri/integration-tests.sh]: Unused?

**Comment by jcvenegas** [comment]
Seems that fedora keeps failing due to https://github.com/kata-containers/tests/pull/1611

James take a look

**Pull request merged**